upload/src/addons/SV/WarningImprovements/Entity/WarningCategory.php

```diff
@@ -134,6 +147,22 @@ protected function _postDelete()
             /** @var \SV\WarningImprovements\XF\Entity\WarningDefinition $warningDefinition */
             if ($warningDefinition->is_custom)
             {
+                if (!empty($this->parent_category_id))
+                {
+                    $newCategoryId = $this->parent_category_id;
+                }
+                else
+                {
+                    $newCategoryId = $this->db()->fetchOne("
+                        SELECT warning_category_id
+                        FROM xf_sv_warning_category
+                        WHERE warning_category_id <> ?
+                    ", $this->warning_category_id);
+                }
+
+                $warningDefinition->sv_warning_category_id = $newCategoryId;
```

Comment:
Should have a sanity check for `$newCategoryId` being empty, and if so block the delete.

Reply:
Check is being made in post delete method

Reply:
You are right, never mind then!